File: Java/DivisibleSumPairs.java
public class DivisibleSumPairs {
    public static int divisibleSumPairs(int n, int k, int[] ar) {
        int count = 0;
        for (int i = 0; i < n; i++) {
            for (int j = i + 1; j < n; j++) {
                if ((ar[i] + ar[j]) % k == 0) {
                    count++;
                }
            }
        }
        return count;
    }

    public static void main(String[] args) {
        int[] ar = {1, 3, 2, 6, 1, 2};
        int k = 3;
        int count = divisibleSumPairs(ar.length, k, ar);
        System.out.println("Number of pairs divisible by " + k + ": " + count);
    }
}

File: Python/DivisibleSumPairs.py
def divisible_sum_pairs(n, k, ar):
    count = 0
    for i in range(n):
        for j in range(i+1, n):
            if (ar[i] + ar[j]) % k == 0:
                count += 1
    return count

# Example usage
ar = [1, 3, 2, 6, 1, 2]
k = 3
count = divisible_sum_pairs(len(ar), k, ar)
print(f"Number of pairs divisible by {k}: {count}")